v3/gssapi/client.go

```diff
@@ -212,3 +216,48 @@ func (client *Client) NegotiateSaslAuth(input []byte, authzid string) ([]byte, e
 
 	return output, nil
 }
+
+func UnmarshalWrapToken(wt *gssapi.WrapToken, data []byte, expectFromAcceptor bool) error {
```

Comment:
I appreciate this new function that appears to do more robust checking while unmarshalling the input, however:

- We should not export the function (and thereby add to our API), unless absolutely necessary.
- I would be much more comfortable if there were some unit tests asserting that this function actually works as intended.